requests/adapters.py

```diff
@@ -420,7 +420,9 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                     preload_content=False,
                     decode_content=False,
                     retries=self.max_retries,
-                    timeout=timeout
+                    timeout=timeout,
+                    enforce_content_length=True,
+                    request_method=request.method
```

Comment:
Unfortunately yes, there's specific logic for a HEAD request in `length_remaining` and it's not accessible from the response object in urllib3. We're passing this information in manually in urllib3 too but it's done at a higher level than `urlopen`.